eval.go

```diff
@@ -821,3 +830,66 @@ func (d *indexData) regexpToMatchTreeRecursive(r *syntax.Regexp, minTextSize int
 	}
 	return &bruteForceMatchTree{}, false, false, nil
 }
+
+// Limits the number of matches on the SearchResult by the given positive limit,
+// returning the remaining limit, if any.
+func (result *SearchResult) LimitMatches(limit int, chunkMatches bool) int {
+	if chunkMatches {
+		for i := range result.Files {
+			limit = limitChunkMatches(&result.Files[i], limit)
+			if limit == 0 {
+				if i != len(result.Files)-1 {
+					result.Files = result.Files[:i+1]
+				}
+				return 0
+			}
+		}
+	} else {
+		for i := range result.Files {
+			limit = limitLineMatches(&result.Files[i], limit)
+			if limit == 0 {
+				if i != len(result.Files)-1 {
+					result.Files = result.Files[:i+1]
+				}
+				return 0
+			}
+		}
+	}
+	return limit
+}
+
+// Limit the number of ChunkMatches in the given FileMatch, returning the
+// remaining limit, if any.
+func limitChunkMatches(file *FileMatch, limit int) int {
+	for i := range file.ChunkMatches {
+		cm := &file.ChunkMatches[i]
+		if len(cm.Ranges) > limit {
+			cm.Ranges = cm.Ranges[:limit]
+		}
+		if len(cm.Ranges) >= limit {
```

Comment:
This pattern appears in a couple of places. What is the motivation for using `>=` instead of `==`?

Reply:
No good reason, I can change it.  I think it ended up this way after I did some refactoring to an earlier approach.